### DISCUSSION TITLE
sb3 is much slower to train than rl_games

### DISCUSSION THREAD

**Discussion by LyuJZ** [isaac-sim/IsaacLab#528] [Q&A]
Hi, I am training a policy for object grasping with KUKA arm with IsaacLab. After the same training iterations, I found that the training results of sb3 and rl_games are similar. However, it takes much longer for sb3 than rl_games during the same number of training iterations. 

Is this phenomenon normal? If so, do you have any suggestions for improving SB3's training speed?

**Comment by Mayankm96** [comment]
Hi @LyuJZ,

Yes, this is a known phenomenon. Please check Section V-A of the paper: https://arxiv.org/abs/2301.04195

TL;DR: SB3 expects the data buffers to be in numpy. This leads to an overhead when converting torch tensors that live on GPU to numpy arrays and vice-versa.

**Reply by LyuJZ** [reply]
From the figure, it seems the sb3 performance is worse than rl_games?

**Comment by araffin** [comment]
@LyuJZ a follow-up that might interest you ;) https://github.com/isaac-sim/IsaacLab/discussions/1749
